src/segment_git.go

```diff
@@ -131,12 +146,44 @@ func (g *git) string() string {
 	if g.props.getBool(StatusColorsEnabled, false) {
 		g.SetStatusColor()
 	}
+	enableHyperlink := g.props.getBool(EnableHyperlink, false)
 	buffer := new(bytes.Buffer)
+
+	// url
+	upstream := replaceAllString("/.*", g.repo.upstream, "")
+	g.repo.url = g.getGitCommandOutput("remote", "get-url", upstream)
+
 	// branchName
 	if g.repo.upstream != "" && g.props.getBool(DisplayUpstreamIcon, false) {
-		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol())
+		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol(g.repo.url))
 	}
-	fmt.Fprintf(buffer, "%s", g.repo.HEAD)
+	// default name
+	repoHEAD := g.repo.PRETTYHEAD
+	if g.repo.upstream != "" && enableHyperlink {
+		// parse url to and strips extra info
```

Comment:
this whole logic should be a separate function

Reply:
I think we should handle hyperlink at a global level(https://github.com/JanDeDobbeleer/oh-my-posh3/pull/307#issuecomment-753464224).  
Outputting something like a template and detect it in the ansi renderer.

Reply:
Yes!

Reply:
Now that i’m back to work I have less time but I’ll try do finish it by the end of the week.
Url template
Global hyperlink
Extra: support ssh cloned repo